Matthewericbassett/ch5685/getpoolinfo

mebassett opened the pull request.

Commit:
add - config options for signedtx endpoint, et al.

Commit:
add - poolInfo endpoint that proxies to SMASH.

Commit:
updating readme for /getPoolInfo

SebastienGllmt commented on README.md:
```diff
@@ -102,6 +102,29 @@ Array<{
+                                 description: string,
+                                 ticker: string,
+                                 homepage: string,
+                                 extended: string
```
what is "extended"

mebassett replied:
I don't know.  actually, upon further inspection, the docs are wrong.  `extended` only sometimes appears.  compare 
` curl https://stage-smash.yoroiwallet.com/api/v1/metadata/40a80b5bfc4dc1ba1bf94a071875c78ee0a563dddef724f5ed21b264d3274a07` with ` curl https://stage-smash.yoroiwallet.com/api/v1/metadata/ca7d12decf886e31f5226b5946c62edc81a7e40af95ce7cd6465122e309d5626`

nicarq replied:
It looks like the extra url param for extended info that doesn't fit the onchain structure

SebastienGllmt commented on README.md:
````diff
@@ -102,6 +102,29 @@ Array<{
+
+```js
+{
+  [poolMetaDataHash: string]: {| name: string,
````
I would prefer if the return type was something like this

```
type RemotePoolInfo = {|
  +owner: string, // I think we don't need this in the backend response since it's specified in the key
  +pledge_address: string,

  // from pool metadata (off chain)
  +name?: string,
  +description?: string,
  +ticker?: string,
  +homepage?: string,
|};

{|
  +info: RemotePoolInfo,
  +history: Array<{|
    epoch: number,
    slot: number,
    tx_ordinal: number,
    cert_ordinal: number,
    payload: any, // TODO: how to store this since different networks have different cert types
  |}>,
|};
```

The main important thing is that we want the `history` list. You can make it an empty array for now, but later it should be a list of all the `PoolRegistration` and `PoolRetirement` that match the pool id

mebassett replied:
Noted.  the history bit will take me a little bit of work to do.  this is essentially just a proxy to smash.  the type I've returned is exactly what smash gives me, no change.  I'll need to populate these fields from the postgresql database.

Commit:
mod - changing cert view so poolHashKey is more reliable.

Commit:
mod - moving rowToCert to an exportable location near its types.

Commit:
add - pool history to poolInfo endpoint.

here's any example:
$  curl -X GET -H "Content-type: application/json" -d '{"poolMetaDataHashes":["a0b42a87829a0260f36d7f500b2142d0fd0bbe127c77110870db91340f44fdfc"]}' http://localhost:8082/getPoolInfo | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  4931  100  4840  100    91  10731    201 --:--:-- --:--:-- --:--:-- 10933
```js
{
  "a0b42a87829a0260f36d7f500b2142d0fd0bbe127c77110870db91340f44fdfc": {
    "info": {
      "name": "CAN-ADA.IO",
      "ticker": "MAPLE",
      "description": "Safe/Reliable/Secure - Experienced Canadian IT/software development professional. Let's grow Cardano together!",
      "homepage": "https://can-ada.io",
      "nonce": "1596064742",
      "pledge_address": "e1d33cabe9bc7a7646243c03f062881d06744b3d53983823178973b9b0"
    },
    "history": [
      {
        "epoch": 208,
        "slot": 2760,
        "tx_ordinal": 2,
        "cert_ordinal": 0,
        "payload": {
          "kind": "PoolRegistration",
          "certIndex": 0,
          "poolParams": {
            "operator": "469fbad36a8a68c5f62f3505ecbe6a461ea262ae20b4de8d39ff08ab",
            "vrfKeyHash": "d15e3ddd3b35456c4f19be6184fa505ae88b87f0e243d361b375f5256197e7b4",
            "pledge": 51000000000,
            "cost": 340000000,
            "margin": 0.01,
            "rewardAccount": "e1d33cabe9bc7a7646243c03f062881d06744b3d53983823178973b9b0",
            "poolOwners": [
              "d33cabe9bc7a7646243c03f062881d06744b3d53983823178973b9b0"
            ],
            "relays": [
              {
                "ipv4": "104.197.180.29",
                "ipv6": null,
                "dnsName": null,
                "dnsSrvName": null,
                "port": 55444
              },
              {
                "ipv4": "34.83.211.149",
                "ipv6": null,
                "dnsName": null,
                "dnsSrvName": null,
                "port": 55444
              },
              {
                "ipv4": "35.203.18.48",
                "ipv6": null,
                "dnsName": null,
                "dnsSrvName": null,
                "port": 55444
              },
              {
                "ipv4": null,
                "ipv6": null,
                "dnsName": "relays.can-ada.io",
                "dnsSrvName": null,
                "port": 55444
              },
              {
                "ipv4": null,
                "ipv6": null,
                "dnsName": "relays-can-ada-io.cloudns.cl",
                "dnsSrvName": null,
                "port": 55444
              }
            ],
            "poolMetadata": {
              "url": "https://can-ada.io/maplepool.json",
              "metadataHash": "5918622d00c5d9159feb470e882f4dbecb04dc3df2e999e2f8e32ef0bd77de53"
            }
          }
        }
      },
      {
        "epoch": 208,
        "slot": 3500,
        "tx_ordinal": 6,
        "cert_ordinal": 0,
        "payload": {
          "kind": "PoolRegistration",
          "certIndex": 0,
          "poolParams": {
            "operator": "469fbad36a8a68c5f62f3505ecbe6a461ea262ae20b4de8d39ff08ab",
            "vrfKeyHash": "d15e3ddd3b35456c4f19be6184fa505ae88b87f0e243d361b375f5256197e7b4",
            "pledge": 51000000000,
            "cost": 340000000,
            "margin": 0.01,
            "rewardAccount": "e1d33cabe9bc7a7646243c03f062881d06744b3d53983823178973b9b0",
            "poolOwners": [
              "d33cabe9bc7a7646243c03f062881d06744b3d53983823178973b9b0"
            ],
            "relays": [
              {
                "ipv4": "104.197.180.29",
                "ipv6": null,
                "dnsName": null,
                "dnsSrvName": null,
                "port": 55444
              },
              {
                "ipv4": "34.83.211.149",
                "ipv6": null,
                "dnsName": null,
                "dnsSrvName": null,
                "port": 55444
              },
              {
                "ipv4": "35.203.18.48",
                "ipv6": null,
                "dnsName": null,
                "dnsSrvName": null,
                "port": 55444
              },
              {
                "ipv4": null,
                "ipv6": null,
                "dnsName": "relays.can-ada.io",
                "dnsSrvName": null,
                "port": 55444
              },
              {
                "ipv4": null,
                "ipv6": null,
                "dnsName": "relays-can-ada-io.cloudns.cl",
                "dnsSrvName": null,
                "port": 55444
              }
            ],
            "poolMetadata": {
              "url": "https://can-ada.io/maplepool.json",
              "metadataHash": "5918622d00c5d9159feb470e882f4dbecb04dc3df2e999e2f8e32ef0bd77de53"
            }
          }
        }
      },
      {
        "epoch": 208,
        "slot": 5160,
        "tx_ordinal": 2,
        "cert_ordinal": 0,
        "payload": {
          "kind": "PoolRegistration",
          "certIndex": 0,
          "poolParams": {
            "operator": "469fbad36a8a68c5f62f3505ecbe6a461ea262ae20b4de8d39ff08ab",
            "vrfKeyHash": "d15e3ddd3b35456c4f19be6184fa505ae88b87f0e243d361b375f5256197e7b4",
            "pledge": 51000000000,
            "cost": 340000000,
            "margin": 0.01,
            "rewardAccount": "e1d33cabe9bc7a7646243c03f062881d06744b3d53983823178973b9b0",
            "poolOwners": [
              "d33cabe9bc7a7646243c03f062881d06744b3d53983823178973b9b0"
            ],
            "relays": [
              {
                "ipv4": "104.197.180.29",
                "ipv6": null,
                "dnsName": null,
                "dnsSrvName": null,
                "port": 55444
              },
              {
                "ipv4": "34.83.211.149",
                "ipv6": null,
                "dnsName": null,
                "dnsSrvName": null,
                "port": 55444
              },
              {
                "ipv4": "35.203.18.48",
                "ipv6": null,
                "dnsName": null,
                "dnsSrvName": null,
                "port": 55444
              },
              {
                "ipv4": null,
                "ipv6": null,
                "dnsName": "relays.can-ada.io",
                "dnsSrvName": null,
                "port": 55444
              },
              {
                "ipv4": null,
                "ipv6": null,
                "dnsName": "relays-can-ada-io.cloudns.cl",
                "dnsSrvName": null,
                "port": 55444
              }
            ],
            "poolMetadata": {
              "url": "https://can-ada.io/maplepool.json",
              "metadataHash": "768dd5c2f36f37f2faf77bbf0c90818103a62955888d717b5e1cc67a4a072108"
            }
          }
        }
      },
      {
        "epoch": 208,
        "slot": 5720,
        "tx_ordinal": 5,
        "cert_ordinal": 0,
        "payload": {
          "kind": "PoolRegistration",
          "certIndex": 0,
          "poolParams": {
            "operator": "469fbad36a8a68c5f62f3505ecbe6a461ea262ae20b4de8d39ff08ab",
            "vrfKeyHash": "d15e3ddd3b35456c4f19be6184fa505ae88b87f0e243d361b375f5256197e7b4",
            "pledge": 51000000000,
            "cost": 340000000,
            "margin": 0.01,
            "rewardAccount": "e1d33cabe9bc7a7646243c03f062881d06744b3d53983823178973b9b0",
            "poolOwners": [
              "d33cabe9bc7a7646243c03f062881d06744b3d53983823178973b9b0"
            ],
            "relays": [
              {
                "ipv4": "104.197.180.29",
                "ipv6": null,
                "dnsName": null,
                "dnsSrvName": null,
                "port": 55444
              },
              {
                "ipv4": "34.83.211.149",
                "ipv6": null,
                "dnsName": null,
                "dnsSrvName": null,
                "port": 55444
              },
              {
                "ipv4": "35.203.18.48",
                "ipv6": null,
                "dnsName": null,
                "dnsSrvName": null,
                "port": 55444
              },
              {
                "ipv4": null,
                "ipv6": null,
                "dnsName": "relays.can-ada.io",
                "dnsSrvName": null,
                "port": 55444
              },
              {
                "ipv4": null,
                "ipv6": null,
                "dnsName": "relays-can-ada-io.cloudns.cl",
                "dnsSrvName": null,
                "port": 55444
              }
            ],
            "poolMetadata": {
              "url": "https://can-ada.io/maplepoolmeta.json",
              "metadataHash": "a0b42a87829a0260f36d7f500b2142d0fd0bbe127c77110870db91340f44fdfc"
            }
          }
        }
      },
      {
        "epoch": 208,
        "slot": 14000,
        "tx_ordinal": 3,
        "cert_ordinal": 0,
        "payload": {
          "kind": "PoolRetirement",
          "certIndex": 0,
          "poolKeyHash": "469fbad36a8a68c5f62f3505ecbe6a461ea262ae20b4de8d39ff08ab",
          "epoch": 209
        }
      }
    ]
  }
}
```

Commit:
updating readme

Commit:
Merge branch 'master' into matthewericbassett/ch5685/getpoolinfo

Commit:
bugfix - no poolInfo errors when smash replies with nothing.

SebastienGllmt commented on README.md:
````diff
@@ -102,6 +102,39 @@ Array<{
+
+```js
+{
+  [poolMetaDataHash: string]: {| info: RemotePoolInfo,
````
confused about what is returned if the pool isn't found. It used to be null, right? Now is it an empty info array and empty history?

mebassett replied:
it'll be an empty object, `{}`

SebastienGllmt replied:
Do you mean like this?
```
{
    'df1750df9b2df285fcfb50f4740657a18ee3af42727d410c37b86207': {},
    'aadf1750df9b2df285fcfb50f4740657a18ee3af42727d410c37b862aa': {},
}
```

If so, I would prefer not returning an empty object if possible because JS isn't really good at dealing with empty objects

mebassett replied:
it is now something like:
```js
{
  "a0b42a87829a0260f36d7f500b2142d0fd0bbe127c77110870db91340f4fdca0": {
    "info": {
      "pledge_address": null
    },
    "history": []
  }
}
```

Commit:
mod - Certificate fields should be strings.

Commit:
mod - changing behavior for addresses that don't exist.

Commit:
Merge branch 'master' into matthewericbassett/ch5685/getpoolinfo
